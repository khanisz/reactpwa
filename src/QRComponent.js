import React, { Component } from "react";
import QrReader from "react-qr-reader";

class QRComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      delay: 300,
      result: "Zeskanuj kod QR",
    };
    this.handleScan = this.handleScan.bind(this);
  }
  handleScan(number) {
    var component = this.props.components[number];
    if (component) {
      component.visible = true;
    }
    
    if (number) {
      this.setState({
        result: "Zeskanowano pomyślnie. Pojawił się nowy wpis w dzienniku"
      });
    }
  }
  handleError(err) {
    console.error(err);
  }
  handleVisibility(number) {
    var component = this.props.components[number];
    if (component) {
      component.visible = true;
    }
    
  }

  render() {
    if (this.state.result != "Zeskanuj kod QR") {
      this.handleVisibility(this.state.result);
    }

    return (
      <div>
        <QrReader
          delay={this.state.delay}
          onError={this.handleError}
          onScan={this.handleScan}
          style={{ width: "100%" }}
        />
        <p>{this.state.result}</p>
      </div>
    );
  }
}

export default QRComponent;
