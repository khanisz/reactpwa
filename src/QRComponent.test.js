import React from "react";
import { shallow } from "enzyme";
import QRComponent from "./QRComponent";

describe("QRComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<QRComponent debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<QRComponent />);
    expect(component).toMatchSnapshot();
  });

});
