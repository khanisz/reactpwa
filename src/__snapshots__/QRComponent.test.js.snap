// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRComponent should render correctly in "debug" mode 1`] = `
<div>
  <Reader
    constraints={null}
    delay={300}
    facingMode="environment"
    onError={[Function]}
    onScan={[Function]}
    resolution={600}
    showViewFinder={true}
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <p>
    Zeskanuj kod QR
  </p>
</div>
`;

exports[`QRComponent should render correctly with no props 1`] = `
<div>
  <Reader
    constraints={null}
    delay={300}
    facingMode="environment"
    onError={[Function]}
    onScan={[Function]}
    resolution={600}
    showViewFinder={true}
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <p>
    Zeskanuj kod QR
  </p>
</div>
`;
